app/services/authenticator.ts

```diff
@@ -72,20 +72,30 @@ export default class AuthenticatorService extends Service {
     await this.syncCurrentUser();
   }
 
-  initiateLogin(redirectPath: string | null) {
-    const isFullUrl = redirectPath?.startsWith(`${config.x.backendUrl}`);
-
-    if (isFullUrl) {
-      window.location.href = `${config.x.backendUrl}/login?next=` + redirectPath;
-    } else if (redirectPath) {
-      window.location.href = `${config.x.backendUrl}/login?next=` + encodeURIComponent(`${window.origin}${redirectPath}`);
-    } else if (this.router.currentURL) {
+  initiateLogin(redirectPathOrUrl: string | null = null) {
+    if (redirectPathOrUrl) {
+      this.initiateLoginAndRedirect(redirectPathOrUrl);
+
+      return;
+    }
+
+    if (this.router.currentURL) {
       window.location.href = `${config.x.backendUrl}/login?next=` + encodeURIComponent(`${window.origin}${this.router.currentURL}`);
     } else {
       window.location.href = `${config.x.backendUrl}/login?next=` + encodeURIComponent(window.origin);
     }
   }
 
+  initiateLoginAndRedirect(redirectPathOrUrl: string) {
```

Comment:
```suggestion
  initiateLoginAndRedirectTo(redirectPathOrUrl: string) {
```o

Lightly suggestive of what the argument is, think this'll read better to